test/transmuxer.test.js

```diff
@@ -877,6 +886,104 @@ QUnit.test('drops packets with unknown stream types', function() {
   QUnit.equal(packets.length, 0, 'ignored unknown packets');
 });
 
+QUnit.module('MP2T TSCorrectionStream', {
+  setup: function() {
+    tsCorrectionStream = new TSCorrectionStream();
+    elementaryStream = new ElementaryStream();
+    elementaryStream.pipe(tsCorrectionStream);
+  }
+});
+
+QUnit.test('Correctly parses rollover PTS', function() {
+  var
+    packets = [],
+    packetData = [0x01, 0x02],
+    pesHeadOne = pesHeader(false, Math.pow(2, 33) - 400),
+    pesHeadTwo = pesHeader(false, Math.pow(2, 33) - 100),
+    pesHeadThree = pesHeader(false, Math.pow(2, 33)),
+    pesHeadFour = pesHeader(false, 50);
+
+  tsCorrectionStream.on('data', function(packet) {
+    packets.push(packet);
+  });
+  elementaryStream.push({
+    type: 'pes',
+    streamType: ADTS_STREAM_TYPE,
+    payloadUnitStartIndicator: true,
+    data: new Uint8Array(pesHeadOne.concat(packetData))
+  });
+  elementaryStream.push({
+    type: 'pes',
+    streamType: ADTS_STREAM_TYPE,
+    payloadUnitStartIndicator: true,
+    data: new Uint8Array(pesHeadTwo.concat(packetData))
+  });
+  elementaryStream.push({
+    type: 'pes',
+    streamType: ADTS_STREAM_TYPE,
+    payloadUnitStartIndicator: true,
+    data: new Uint8Array(pesHeadThree.concat(packetData))
+  });
+  elementaryStream.push({
+    type: 'pes',
+    streamType: ADTS_STREAM_TYPE,
+    payloadUnitStartIndicator: true,
+    data: new Uint8Array(pesHeadFour.concat(packetData))
+  });
+  elementaryStream.flush();
+
+  QUnit.equal(packets.length, 4, 'built four packets');
+  QUnit.equal(packets[0].type, 'audio', 'identified audio data');
+  QUnit.equal(packets[0].data.byteLength, packetData.length, 'parsed the correct payload size');
+  QUnit.equal(packets[0].pts, Math.pow(2, 33) - 400, 'correctly parsed the pts value');
+  QUnit.equal(packets[1].pts, Math.pow(2, 33) - 100, 'Does not rollover on minor change');
+  QUnit.equal(packets[2].pts, Math.pow(2, 33), 'correctly parses the max pts value');
+  QUnit.equal(packets[3].pts, Math.pow(2, 33) + 50, 'correctly parsed the rollover pts value');
+});
+
+QUnit.test('Correctly parses multiple PTS rollovers', function() {
+  var
+    packets = [],
+    packetData = [0x01, 0x02],
+    pesArray = [pesHeader(false, 1),
+                pesHeader(false, Math.floor(Math.pow(2, 33) * (1 / 3))),
```

Comment:
Might be nice to simply make a constant for the value 2^33 instead of repeating it or even a function so you can simply do `pesHeader(false, makeTS(1 / 3))`?